types/src/bytesrepr.rs

```diff
@@ -532,45 +532,32 @@ impl<T: FromBytes> FromBytes for VecDeque<T> {
     }
 }
 
-macro_rules! impl_to_from_bytes_for_array {
-    ($($N:literal)+) => {
-        $(
-            impl ToBytes for [u8; $N] {
-                #[inline(always)]
-                fn to_bytes(&self) -> Result<Vec<u8>, Error> {
-                    Ok(self.to_vec())
-                }
-
-                #[inline(always)]
-                fn serialized_length(&self) -> usize { $N }
-
-                #[inline(always)]
-                fn write_bytes(&self, writer: &mut Vec<u8>) -> Result<(), Error> {
-                    writer.extend_from_slice(self);
-                    Ok(())
-                }
-            }
+impl<const COUNT: usize> ToBytes for [u8; COUNT] {
```

Comment:
Yes, it wasn't technically needed, but it made writing [this function](https://github.com/casper-network/casper-node/pull/2980/files#diff-02b034455230e126dacd66a4c54da3bb7ffefa792f92f377b19606075894299aR24-R39) much easier, hence I did the change.

Const generics are available since [Rust 1.51.0 (available from Mar. 25, 2021)](https://github.com/rust-lang/rust/blob/master/RELEASES.md#version-1510-2021-03-25). Our toolchain in `smart_contracts` is `nightly-2022-01-13` so I think we're all set.